File: LeetCode Problems/1 to 100/7. Reverse Integer/7. Reverse Integer.cpp
class Solution {
public:
    int reverse(int x) {
        long long rev_x = 0;
        while (x != 0) {
            int rem = x % 10;
            rev_x = rev_x * 10 + rem;
            x /= 10;
            if (rev_x > INT_MAX || rev_x < INT_MIN)
                return 0;
        }
        return rev_x;
    }
};

File: LeetCode Problems/1 to 100/7. Reverse Integer/7. Reverse Integer.cs
public class Solution {
    public int Reverse(int x) {
        int num = 0;
        bool isNumberNegative = false;
        if (x < 0) {
            isNumberNegative = true;
            x = x * (-1);
        }

        string number = x.ToString();
        var array = number.ToCharArray().Reverse();
        number = string.Join("", array);

        try {
            num = Convert.ToInt32(number);
        } catch (Exception e) {
            num = 0;
        }

        if (isNumberNegative) num = num * (-1);

        return num;
    }
}
